### PR TITLE
#6Task.oleg.slucki

### PR DESCRIPTION
**Pull request by bref987**
https://travis-ci.com/github/bref987/js-assignments/builds/174407424

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -82,7 +91,8 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    const perimetre = a + b + c;
+    return [a, b, c].filter(a => a < perimetre - a).length === 3 ? true : false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Actually, we don't need ternary operator here
just return [a, b, c].filter(a => a < perimetre - a).length === 3
Please, see https://javascript.info/logical-operators

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -251,7 +282,19 @@ function reverseInteger(num) {
...
+        let cardNum = parseInt(ccn[i]);
+        if ((ccn.length - i) % 2 === 0) {
+            cardNum *= 2;
+            if (cardNum > 9) {
+                cardNum -= 9;
+            }
+        }
+        sum += cardNum;
+    }
+    return sum % 10 == 0;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator === is always preferable
sometimes ''==' may cause an error
(check at console :
"0" == 0
and "0" === 0)"

**Comment by vramaniuk** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js (hunk 2 above)
Comment (bref987) [reply task/06-conditions-n-loops-tasks.js]: ==  (by value) replaced by === (by type and value)
I see and understand the difference

File: task/06-conditions-n-loops-tasks.js (hunk 1 above)
Comment (bref987) [reply task/06-conditions-n-loops-tasks.js]: useless ternary operator is removed
logical-operators article is read

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -30,7 +30,8 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    return num % 3 === 0 && num % 5 === 0 ?
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: In this case, the implementation via "if" "return" is much better. Several ternary operators are not very readable.

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -296,7 +342,27 @@ function getDigitalRoot(num) {
...
+        '{':'}',
+        '(':')',
+        '<':'>'
+    };
+    const stack = [];
+    const arrStrBalance = str.split('');
+
+    if (arrStrBalance.length % 2 !== 0) {
+        return false;
+    } else {
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" after return is redundant

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -332,7 +398,29 @@ function isBracketsBalanced(str) {
...
+    const msec = difDate.getTime();
+    const sec = msec / 1000;
+    const min = sec / 60;
+    const hour = min / 60;
+    const day = hour / 24;
+    const month = day / 30;
+    const year = month / 12;
+
+    if (sec <= 45) {return `a few seconds ago`}
+    else if (sec <= 90) {return `a minute ago`}
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js:421]: "else" after return is redundant

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js (hunk 3 above)
Comment (bref987) [reply task/06-conditions-n-loops-tasks.js]: refactor to 'if - else if' contruction for better readabitilty

File: task/06-conditions-n-loops-tasks.js (hunk 4 above)
Comment (bref987) [reply task/06-conditions-n-loops-tasks.js]: remove redandant 'else'

File: task/06-conditions-n-loops-tasks.js (hunk 5 above)
Comment (bref987) [reply task/06-conditions-n-loops-tasks.js:421]: remove redundant 'else'

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**